doc/source/serve/architecture.md

```diff
@@ -24,62 +24,89 @@ There are three kinds of actors that are created to make up a Serve instance:
   by the replica using `@serve.batch`, see the [batching](serve-batching) docs).
 
 ## Lifetime of a Request
 
-When an HTTP request is sent to the router, the follow things happen:
+When an HTTP request is sent to the HTTP proxy, the following things happen:
 
-- The HTTP request is received and parsed.
-- The correct deployment associated with the HTTP url path is looked up. The
+1. The HTTP request is received and parsed.
+2. The correct deployment associated with the HTTP URL path is looked up. The
   request is placed on a queue.
-- For each request in a deployment queue, an available replica is looked up
-  and the request is sent to it. If there are no available replicas (there
-  are more than `max_concurrent_queries` requests outstanding), the request
-  is left in the queue until an outstanding request is finished.
+3. For each request in a deployment's queue, an available replica is looked up in round-robin fashion
+  and the request is sent to it. If there are no available replicas (i.e. there
+  are more than `max_concurrent_queries` requests outstanding at each replica), the request
+  is left in the queue until a replica becomes available.
 
-Each replica maintains a queue of requests and executes one at a time, possibly
-using asyncio to process them concurrently. If the handler (the function for the
-deployment or `__call__`) is `async`, the replica will not wait for the
-handler to run; otherwise, the replica will block until the handler returns.
+Each replica maintains a queue of requests and executes requests one at a time, possibly
+using `asyncio` to process them concurrently. If the handler (the deployment function or the `__call__` method of the deployment class) is declared with `async def`, the replica will not wait for the
+handler to run.  Otherwise, the replica will block until the handler returns.
 
-## FAQ
+When making a request via [ServeHandle](serve-handle-explainer) instead of HTTP, the request is placed on a queue in the ServeHandle, and we skip to step 3 above.
 
 (serve-ft-detail)=
 
-### How does Serve handle fault tolerance?
+## Fault tolerance
 
 Application errors like exceptions in your model evaluation code are caught and
 wrapped. A 500 status code will be returned with the traceback information. The
 replica will be able to continue to handle requests.
 
-Machine errors and faults will be handled by Ray. Serve utilizes the [actor
-reconstruction](actor-fault-tolerance) capability. For example, when a machine hosting any of the
-actors crashes, those actors will be automatically restarted on another
+Machine errors and faults will be handled by Ray Serve as follows:
+
+- When replica actors fail, the Controller actor will replace them with new ones.
+- When the HTTP proxy actor fails, the Controller actor will restart it.
+- When the Controller actor fails, Ray will restart it.
+- When the Ray cluster fails, Ray Serve cannot recover.
```

Comment:
Ray Serve replies on Kubernetes to recover.

Reply:
please update that in kubernetes the operator will recover it.

also add a note that GCS HA can be used to avoid the cluster failure. ask @iycheng where we can link to relevant docs.

Reply:
Does the kubernetes note apply just to failed worker nodes, or to a failed cluster?  Also, this is referring specifically to Kuberay, right?

Reply:
A failed cluster as well. And yes, specific to kuberay.